### DISCUSSION TITLE
Question: Setting Latency for output.srt in Liquidsoap

### DISCUSSION THREAD

**Discussion by esraell** [savonet/liquidsoap#4256] [Q&A] [answered]
Hey all,

I'm looking for a way to set latency for the `output.srt` function in Liquidsoap. When using FFmpeg, I know you can easily configure latency using the `?latency=100` parameter in the SRT URL. However, I'm not sure how to achieve the same thing in Liquidsoap.

I’ve seen that `output.srt` supports a `socket_options` parameter, which includes options like `SRTO_LATENCY`. For example:

```
output.srt(
  live_source_1,
  mode="caller",
  host="example.com",
  port=1234,
  socket_options=[
    ("SRTO_LATENCY", 2000)  # Set latency to 2000ms
  ]
)
```

However, I’m running into issues where this configuration doesn’t seem to take effect or throws errors. Is there a recommended way to set latency, or does Liquidsoap handle it differently?

Would appreciate any guidance or examples!

Thanks!

**Comment by toots** [comment, marked as answer]
Hi! I think that your problem here is that this configuration option does not exist! I imagine someone (or something!) must have hallucinated it.. 😄

The source of truth for our documentation is our online API reference. Here's the link for `output.srt` with all the options: https://www.liquidsoap.info/doc-2.3.0/reference.html#output.srt

You can also use the command line:
```shell
% liquidsoap -h output.srt
...
```

I'm looking at the doc and it does not appear that we support setting SRT sockets options at the moment. I'm creating a ticket for it, I think it'll be a nice new feature to have.

**Reply by esraell** [reply]
That would be awesome, thanks!

**Reply by esraell** [reply]
@toots Would it be possible to add a latency parameter for SRT? This would be really helpful.

**Reply by toots** [reply]
@esraell Check out: https://github.com/savonet/liquidsoap/pull/4325 !

**Reply by toots** [reply]
I think that this could go out with `2.3.1` in early feb.
